dbt/adapters/duckdb/utils.py

```diff
@@ -47,7 +48,7 @@ def as_dict(self) -> Dict[str, Any]:
         return base
 
     @classmethod
-    def create_from_source(cls, source: SourceDefinition) -> "SourceConfig":
+    def create_from_source(cls, source: Any) -> "SourceConfig":
```

Comment:
Heard - let me bug @MichelleArk & @colin-rogers-dbt about this and get back to you

Reply:
I went ahead with option 3 above in dbt-core: https://github.com/dbt-labs/dbt-core/issues/9766, since it was actually pretty straightforward to do in a backward-compatible way (meta merges table.meta and source.meta, with table.meta taking precedence), and updated the branch to remove the `source_meta` reference in favor of just `meta` which now includes both table + source meta attributes. 

I also updated the reference to `config._extra` to `config.extra`, as those should be identical and `extra` is already part of the RelationConfig protocol: https://github.com/dbt-labs/dbt-adapters/blob/35bd3629c390cf87a0e52d999679cc5e33f36c8f/dbt/adapters/contracts/relation.py#L41.

Tests were passing for me locally against dbt-core@main. @jwills -- could we kick off CI again?

Reply:
Looks great @MichelleArk , thank you!